### PR TITLE
Bucket assets destroy hook

### PR DESCRIPTION
**Pull request by mvandervliet**
Full automation for Object Storage bucket images. Process is as follows:

1. Provision bucket from OSB `helm install muservices provision/ -f myvalues.yaml`
    > Bucket provisioned and `PAR` secret binding created
1. Install mushop with helm `helm install mushop mushop/ -f myvalues.yaml`
    > Bucket populated with product images
1. Verify images in bucket (CLI or console)
1. Delete mushop with helm `helm del mushop`
1. `mushop-assets-destroy` Job created via `"helm.sh/hook": pre-delete`
    > Bucket emptied (entirely)
1. Deprovision bucket from the provision installation `helm del muservices`
    > Bucket deleted
1. Verify all service instances destroyed `kubectl get serviceinstances`

### CODE REVIEW

File: src/assets/deploy.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
+/**
+ * Copyright © 2020, Oracle and/or its affiliates. All rights reserved.
+ * The Universal Permissive License (UPL), Version 1.0
+ */
 const fs = require('fs');
 const path = require('path');
 const mime = require('mime');
-const axios = require('axios');
```
Comment (junior) [inline src/assets/deploy.js]: :)

File: src/assets/package.json [hunk 2]
```diff
@@ -12,9 +12,10 @@
     "deploy": "node deploy.js"
   },
   "dependencies": {
-    "axios": "^0.19.0",
```
Comment (junior) [inline src/assets/package.json]: :)

File: deploy/complete/helm-chart/mushop/charts/assets/templates/destroy-job.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "assets.fullname" . }}-destroy
+  labels: {{ include "assets.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  ttlSecondsAfterFinished: 60
```
Comment (junior) [inline deploy/complete/helm-chart/mushop/charts/assets/templates/destroy-job.yaml]: The ttl is working for this job and correctly removing/recycling the pod after finished?

Comment (mvandervliet) [reply deploy/complete/helm-chart/mushop/charts/assets/templates/destroy-job.yaml]: It is cleaning, yes.

**Comment by mvandervliet** [comment]
> How does the assets chart affect mocked deployments ?

`assets` service is always configured, and serves image files from the local build. Only when the `api` service is configured with the `STATIC_MEDIA_URL` will the service be bypassed. The additional functionality here is that `assets` has a job to deploy assets to OOS when a PAR is given (and destroy when uninstalled)

**Pull request merged**